$MirOS$

	#1: Fix CVE-2011-3389; suggested for upstream inclusion.
	#2: Fix spelling; suggested for upstream inclusion.

--- WWW/Library/Implementation/HTTP.c.orig	Mon Jun 13 00:18:30 2011
+++ WWW/Library/Implementation/HTTP.c	Sat Mar  3 18:20:21 2012
@@ -117,9 +117,15 @@ SSL *HTGetSSLHandle(void)
 	ssl_ctx = SSL_CTX_new();
 	X509_set_default_verify_paths(ssl_ctx->cert);
 #else
+	long ssl_opts;
+
+	ssl_opts = SSL_OP_ALL;
+#ifdef SSL_OP_DONT_INSERT_EMPTY_FRAGMENTS
+	ssl_opts &= ~SSL_OP_DONT_INSERT_EMPTY_FRAGMENTS;
+#endif
 	SSLeay_add_ssl_algorithms();
 	ssl_ctx = SSL_CTX_new(SSLv23_client_method());
-	SSL_CTX_set_options(ssl_ctx, SSL_OP_ALL);
+	SSL_CTX_set_options(ssl_ctx, ssl_opts);
 	SSL_CTX_set_default_verify_paths(ssl_ctx);
 	SSL_CTX_set_verify(ssl_ctx, SSL_VERIFY_PEER, HTSSLCallback);
 #endif /* SSLEAY_VERSION_NUMBER < 0x0800 */
@@ -1991,7 +1997,7 @@ static int HTLoadHTTP(const char *arg,
 		 * interactive user, with option to use 303 for 301 (but not
 		 * for 307), and otherwise refuse the redirection.  We also
 		 * don't allow permanent redirection if we keep POST content.
-		 * If we don't find the Location header or it's value is
+		 * If we don't find the Location header or its value is
 		 * zero-length, we display whatever the server returned, and
 		 * the user should RELOAD that to try again, or make a
 		 * selection from it if it contains links, or Left-Arrow to the
